Bugzilla Wrapper Notebook

malialiu opened the pull request:
Showcase of bugzilla wrapper using perceval with traditional and rest API backend.
The following functions are include in the showcase:
- download_bugzilla
- parse_bugzilla
- parse_bugzillarest

carlosparadis commented on vignettes/bugzilla_wrapper_showcase.Rmd:
```diff
@@ -0,0 +1,93 @@
+  %\VignetteEncoding{UTF-8}
+---
+# Introduction
+This notebook explains how to fetch bugs from Bugzilla from bugzilla site by download the bugzilla issues and comments using perceval. We will be using three function which can be found in the R folder.
```
function**s**

carlosparadis commented on vignettes/bugzilla_wrapper_showcase.Rmd:
```diff
@@ -0,0 +1,93 @@
+  ~/kaiaulu/R/download.R: download_bugzilla
+  ~/kaiaulu/R/parser.R: parse_bugzilla, parse_bugzillarest
+
+Note: The perceval has two endpoint for bugzilla, so we have two function to parse bugzilla data with different endpoint. One is the traditional backend, and the other one is the REST API backend for Buzilla 5.0 (or higher) servers. The REST API backend will getting more information of issues creator and issue assignee than the traditional backend. For more information, please check: https://gith ...
```
functions 

be getting

carlosparadis commented on vignettes/bugzilla_wrapper_showcase.Rmd:
````diff
@@ -0,0 +1,93 @@
+```
+
+# Project Configuration File (Parameters Needed)
+The parameters necessary for analysis are kept in a project configuration file to ensure reproducibility. In this project, we will using the perceval path which is keep in the  tools.yml in this example to parse the bugzilla data. 
+
+Note: you will need to modify the perceval path in tools.yml to your perceval path, you can find your perceval path by running the following command: 
+  Mac and Ubuntu: which perceval
+  Windows: where perceval
````
Let's move this to the troubleshooting section of the wiki. It is fine to explain the parameters being used, but for the more general files that are used everywhere (e.g. what is a project configuration file or how tools.yml works) this is best left on the wiki or the README.md to avoid repeating ourselves everywhere.

carlosparadis commented on vignettes/bugzilla_wrapper_showcase.Rmd:
````diff
@@ -0,0 +1,93 @@
+perceval_path <- tool[["perceval"]]
+```
+
+# Download Bugzilla issues and comments using Perceval in traditional Bugzilla backend
````
This header is too long. Let's call it # Traditional Backend.

carlosparadis commented:
Hi Malia,

I think this PR is close to done, however:

- Please make shorter headers. If necessary, use sub-headers. 
- Run your english portion of the Notebook on grammarly.com. It is missing some plurals and articles in between. Also, ask @leilani-reich and @nicoelee123 to grammar check it after you run through Grammarly. They can do so after you make a new commit, and can comment directly in line as I have been doing. @nicoelee123 @leilani-reich when you are done reviewing, please post a message to the PR with an 👍 Malia can then send another commit, then I will make a final pass.
- You should move some of the information to the project configuration file. Please make a new one for Red Hat, and extend this portion on the RedHat project configuration file: 

https://github.com/sailuh/kaiaulu/blob/716a46e546d5293f3b2856cace62311c95aae977/conf/tomcat.yml#L70-L71

Include the red hat config in your next commit with your suggestions of fields and I will review it.

carlosparadis commented:
Also, please git squash after the changes are performed, since it is hard to review as is due to the mix of commits with Leilani.

carlosparadis commented:
@leilani-reich Thanks!

@malialiu I see a large number of conflicts here which blocks me from accepting any PR, so make sure you also fix them for the PR Saturday deadline :)

leilani-reich commented:
@malialiu 

Hi Malia, I believe these are the asks @carlosparadis mentioned:

1. Make sure for each parser function section, you add an example link of an issue from an actual Bugzilla website.
For instance, for RedHat bugzilla, we could link https://bugzilla.redhat.com/show_bug.cgi?id=2043044 based on an issue that the user sees in the parsed bugzilla table. This is a good way for the user to sanity check that the parsed bugzilla tables have the correct output.

2. Change the Bugzilla website currently being used for the Perceval traditional Bugzilla backend. We are using RedHat bugzilla now, but you should find a website that does not use the Bugzilla REST API and only the traditional Bugzilla backend.

3. Run all the functions on the notebook and compare the parsed bugzilla tables. Check if they match in terms of the number of rows and names of the columns. Check if any column is missing from one table that is in another.

I think those were the 3 main asks. Actually, from the meeting I think there were more things that came up. I'll add here.

4. Make the headers shorter for each of the sections in the Bugzilla notebook.  So for example, instead of "## Download Bugzilla issues and comments from REST API", we could use "REST API Downloader".

5. Add @seealso and @return parameters for the Bugzilla functions in the function descriptions. For @seealso, list the functions that are related to the function and make a short sentence to explain how the function is related. For @return, write what is being returned from the function.

Once you have finished your revisions, we want to send an email to Carlos on the order to accept our Bugzilla-related pull requests. I think it would start with my Bugzilla pull requests first, then your parser pull request, and finally your Bugzilla showcase pull request last.

@carlosparadis Please let me know if I missed anything.

carlosparadis commented:
@leilani-reich Yes this is all good. @malialiu  Reminder for Saturday you don't need to do all of them, but at least commit some. The more you can do, the better since your next and last week is basically finishing this. 

Thanks for summarizing this Leilani.

carlosparadis commented:
@leilani-reich @malialiu I just remembered one more thing. The NAs. The first comment on an issue appears in a separate column as issue description. I am not clear if this applies to all 3 downloaders, and also, this should be included in the docs of all functions if so.

malialiu commented:
> 1. Make sure for each parser function section, you add an example link of an issue from an actual Bugzilla website.
For instance, for RedHat bugzilla, we could link https://bugzilla.redhat.com/show_bug.cgi?id=2043044 based on an issue that the user sees in the parsed bugzilla table. This is a good way for the user to sanity check that the parsed bugzilla tables have the correct output.
>
I added. Let me know if I missed anything.

> 2. Change the Bugzilla website currently being used for the Perceval traditional Bugzilla backend. We are using RedHat bugzilla now, but you should find a website that does not use the Bugzilla REST API and only the traditional Bugzilla backend.
>
This site only use perceval traditional bugzilla backend: https://bugzilla.samba.org/

>3. Run all the functions on the notebook and compare the parsed bugzilla tables. Check if they match in terms of the number of rows and names of the columns. Check if any column is missing from one table that is in another.
>
![image](https://user-images.githubusercontent.com/76604944/233766530-102344df-8f2d-4603-ad90-563556c22ffd.png)

> 4. Make the headers shorter for each of the sections in the Bugzilla notebook. So for example, instead of "## Download Bugzilla issues and comments from REST API", we could use "REST API Downloader".
>
I changed to as followings;
![image](https://user-images.githubusercontent.com/76604944/233766607-25ba47a8-eedd-48bd-b82d-b7b9c04100ad.png)

> 5. Add @Seealso and @return parameters for the Bugzilla functions in the function descriptions. For @Seealso, list the functions that are related to the function and make a short sentence to explain how the function is related. For @return, write what is being returned from the function.
>
I added them to this pull request, but I believe that this pull request is only for the showcase. Should I add them to the function pull request to avoid any interference with the function contribution?

carlosparadis commented:
All looks good! 

> I added them to this pull request, but I believe that this pull request is only for the showcase. Should I add them to the function pull request to avoid any interference with the function contribution?

Yes, add to the other pull requests as planned :)

malialiu commented:
Hi Carlos,

If we change our parser function to make the first comment to issue description column, but we will still download the issue description as comment from the downloader function, should I mention that in this notebook?

carlosparadis commented:
Reply to: https://github.com/sailuh/kaiaulu/pull/164#issuecomment-1518574524

@malialiu Yes. These kinds of edge cases are especially important to be mentioned in the Notebook, but it should also be documented in your function too. However, I am still confused on whether we need this separation or not. I need to see the issue example so I can discern if it makes sense to separate this first comment into a separate column.

carlosparadis commented:
Don't worry too much about the codcov comment. The important thing is that the checks pass, which I see these are. :)

leilani-reich reviewed (approved):
Looks good!

carlosparadis commented:
@malialiu wow that was quick! Thanks for resolving the last merge conflict. Just waiting on the checks to review.

carlosparadis commented:
@malialiu https://github.com/sailuh/kaiaulu/pull/164#issuecomment-1518531441

I almost missed this. Could you attach this spreadsheet that compares everything? I will transfer it over to wiki down the road so having the actual file would save me a lot of time. Thank you for putting this together!

malialiu commented:
Attached is the spreadsheet. [Bugzilla Checks.xlsx](https://github.com/sailuh/kaiaulu/files/11362775/Bugzilla.Checks.xlsx)

carlosparadis reviewed (approved):
Looks good and passes the checks.

malialiu commented:
> Attached is the spreadsheet. [Bugzilla Checks.xlsx](https://github.com/sailuh/kaiaulu/files/11362775/Bugzilla.Checks.xlsx)

Sorry, I sent the wrong one, this is the updated one, but I'm wondering that even issue description is not shown on the table, it is shown on comment_body column for the crawler function.. 
[Bugzilla Checks (1).xlsx](https://github.com/sailuh/kaiaulu/files/11362817/Bugzilla.Checks.1.xlsx)

carlosparadis commented:
@malialiu 

> Sorry, I sent the wrong one, this is the updated one, but I'm wondering that even issue description is not shown on the table, it is shown on comment_body column for the crawler function..
[Bugzilla Checks (1).xlsx](https://github.com/sailuh/kaiaulu/files/11362817/Bugzilla.Checks.1.xlsx)

Just to be clear: You implemented the specification of where the issue comment should be according to https://github.com/sailuh/kaiaulu/pull/155#issuecomment-1519036410 right? 

I am not sure I understand the comment above. Are you referring to the correctness of the .xlsx?
